diff --git a/dist/index.css b/dist/index.css
index 8276289d55abd0f70d399e33c5f6c2d006745000..c70f12f1644dcebea6730b9b201a6933fa28969b 100644
--- a/dist/index.css
+++ b/dist/index.css
@@ -9,8 +9,6 @@
 
 .ck-content pre {
 	padding: 1em;
-	color: hsl(0, 0%, 20.8%);
-	background: hsla(0, 0%, 78%, 0.3);
 	border: 1px solid hsl(0, 0%, 77%);
 	border-radius: 2px;
 
@@ -28,6 +26,11 @@
 	min-width: 200px;
 }
 
+.ck-content pre:not(.hljs) {
+	color: hsl(0, 0%, 20.8%);
+	background: hsla(0, 0%, 78%, 0.3);
+}
+
 .ck-content pre code {
 		background: unset;
 		padding: 0;
diff --git a/dist/index.js b/dist/index.js
index f0f8aec2323e8cdb34fa0bb4f48c2b433abcb804..b9600d356df2bf1922d51c6cebd1f74ee3ed9155 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -656,6 +656,7 @@ function getLastOutdentableSequenceRange(model, position, sequence) {
         // Attributes added only in the editing view.
         if (useLabels) {
             preAttributes['data-language'] = languagesToLabels[codeBlockLanguage];
+            preAttributes.class = "hljs";
             preAttributes.spellcheck = 'false';
         }
         const codeAttributes = languagesToClasses[codeBlockLanguage] ? {
